public/docs/ts/latest/guide/index.jade

```diff
@@ -1,10 +1,501 @@
+  There is no evidence of a framework, no "Angular" in it at all.
+  
+  In fact, it really is *just a class*. It's not a component until we *tell Angular about it*.
+  
+  We tell Angular that `HeroListComponent` is a component by attaching **metadata** to the class.
+  
+  The easy way to attach metadata in TypeScript is with a **decorator**. 
+  Here's some metadata for `HeroListComponent`:
++makeExample('intro/ts/app/hero-list.component.ts', 'metadata')
+:marked
+  Here we see the `@Component` decorator which (no surprise) identifies the class
+  immediately below it as a Component class.
+  
+  A decorator is a function. Decorators often have a configuration parameter. 
+  The `@Component` decorator takes a required configuration object with the
+  information Angular needs to create and present the component and its view.
+  
+  Here we see a few of the possible `@Component` configuration options:
+  
+  * `selector` - a css selector that tells Angular to create and insert an instance of this component 
+  where it finds a `<hero-list>` tag in *parent* HTML. 
+  If the template of the application shell (a Component) contained
+<div style="margin-left:30px">
+code-example(language="html").
+    &lt;hero-list>&lt;/hero-list>
+</div>  
+:marked
+  >Angular would replace that tag with an instance of the `HeroListComponent` view.
+  
+  * `templateUrl` - the address of this component's template which we showed [above](#the-template).
+  
+  * `directives` - an array of the Components or Directives that *this* template requires.
+  We saw in the last line of our template that we expect Angular to insert a `HeroDetailComponent`
+  in the space indicated by `<hero-detail>` tags. 
+  Angular will do so only if we mention the `HeroDetailComponent` in this `directives` array.
+  
+  * `providers` - an array of **dependency injection providers** for services that the component requires.
+  This is one way to tell Angular that our component's constructor requires a `HeroService` 
+  so it can get the list of heroes to display. We'll get to dependency injection in a moment.
+figure
+  img(src="/resources/images/devguide/intro/template-metadata-component.png" alt="Metadata" align="left" style="height:200px; margin-left:-40px;margin-right:10px" )
+:marked
+  The `@Component` function takes the configuration object and turns it into metadata that it attaches
+  to the component class definition. Angular discovers this metadata at runtime and thus knows how to do "the right thing".
+  
+  The template, metadata, and component together describe the view.
+
+  We apply other metadata decorators in a similar fashion to guide Angular behavior. 
+  The `@Injectable`, `@Input`, `@Output`, `@RouterConfig` are a few of the more popular decorators
+  we'll master as our Angular knowledge grows.
+<br clear="all">
+:marked
+  The architectural take-away is that we must add metadata to our JavaScript code
```

Comment:
Maybe we should remove "JavaScript" here and just talk about adding metadata to code? Technically JavaScript doesn't support decorators, not sure if this would confuse ppl. Because on the other hand, we don't mention anything about ES5 code until this point.

I think it's totally legit to say sth. like "decorators allow us to add metadata to our code".